akka-http-core/src/main/scala/akka/http/impl/engine/ws/Utf8Encoder.scala

```diff
@@ -48,7 +48,7 @@ private[http] object Utf8Encoder extends GraphStage[FlowShape[String, ByteString
             b(0xc0 | ((char & 0x7c0) >> 6)) // upper 5 bits
             b(0x80 | (char & 0x3f)) // lower 6 bits
           } else if (char >= SurrogateFirst && char < SurrogateSecond)
-            surrogateValue = 0x10000 | ((char ^ SurrogateFirst) << 10)
+            surrogateValue = 0x10000 + ((char & 0x3ff) << 10)
```

Comment:
👍 matches the example conversion code here: http://unicodebook.readthedocs.io/unicode_encodings.html#utf-16-surrogate-pairs

Two possible improvements to readability would be to consistently use curly braces to more easily distinguish the nested if-else, and rename to use high/low instead of first/second.

Reply:
Thanks, @jonas. Good suggestions.